src/compose/volume.ts

```diff
@@ -17,34 +17,28 @@ export interface VolumeConstructOpts {
 
 export interface VolumeConfig {
 	labels: LabelObject;
+	driver: string;
 	driverOpts: Docker.VolumeInspectInfo['Options'];
 }
 
 export interface ComposeVolumeConfig {
+	driver: string;
 	driver_opts: Dictionary<string>;
 	labels: LabelObject;
 }
 
 export class Volume {
-	public appId: number;
-	public name: string;
-	public config: VolumeConfig;
-
 	private logger: Logger;
 	private docker: Docker;
 
 	private constructor(
-		name: string,
-		appId: number,
-		config: VolumeConfig,
+		public name: string,
+		public appId: number,
+		public config: VolumeConfig,
```

Comment:
Simplified the fact these are passed in, and are publicly exposed on the instance 👍